sdp/njit_sdp.py

```diff
@@ -17,9 +17,9 @@ def _sliding_dot_product(Q, T):
     return out
 
 
-Q = np.random.rand(50)
-T = np.random.rand(100)
-_sliding_dot_product(Q, T)
+def setup(Q, T):
+    _sliding_dot_product(np.random.rand(50), np.random.rand(100))
```

Comment:
In `test.py`, I noticed that the function `setup` is executed inside the `for _ in range(n_iter)` loop. Not sure why though... if there is a certain reason behind running it again and again within that loop, then maybe we should try to reduce its running time by choosing smaller arrays like length 3 & 10 for Q & T, respectively.

Reply:
No, it should be moved outside. Though, it is absolutely needed for `pyFFTW` since it needs time to setup arrays for each different size input. `njit` does not suffer from this problem.

Reply:
> Though, the the setup call is absolutely needed for pyFFTW since it needs time to setup arrays for each different input size. 


Ahh.. interesting! Now that you mentioned it, I can see that the `setup` function runs sliding dot product for Q and T. 

> njit does not suffer from this problem.

Got it. The only thing that I just noticed here is that the function `setup` in `njit_sdp.py` takes `Q` and `T` as input but then does NOT run the sliding dot product on those arrays. I know it is okay from njit perspective to not run it for different lengths but just wanted to highlight it here. If you are planning to be consistent for th function `setup`, then please ignore this message. Otherwise, we may remove the params Q and  T here.

Reply:
Yes, I wanted it to be consistent and for all `setup` functions to have the exact same signature